test/unittesting/config_test.go

```diff
@@ -0,0 +1,1403 @@
+	cookieSameSite0 := " Lax "
+	configValue := supertokens.TypeInput{
+		Supertokens: &supertokens.ConnectionInfo{
+			ConnectionURI: "http://localhost:8080",
+		},
+		AppInfo: supertokens.AppInfo{
+			APIDomain:     "api.supertokens.io",
+			AppName:       "SuperTokens",
+			WebsiteDomain: "supertokens.io",
+		},
+		RecipeList: []supertokens.Recipe{
+			session.Init(&sessmodels.TypeInput{
+				CookieSameSite: &cookieSameSite0,
+			}),
+		},
+	}
+	BeforeEach()
+	StartUpST("localhost", "8080")
+	err := supertokens.Init(configValue)
+	if err != nil {
+		t.Errorf(err.Error())
+	}
+
+	defer AfterEach()
+}
+
+func TestSuperTokensInitWithConfigForSessionModulesWithSameSiteValueAsNone(t *testing.T) {
+	cookieSameSite0 := "None "
+	configValue := supertokens.TypeInput{
+		Supertokens: &supertokens.ConnectionInfo{
+			ConnectionURI: "http://localhost:8080",
+		},
+		AppInfo: supertokens.AppInfo{
+			APIDomain:     "api.supertokens.io",
+			AppName:       "SuperTokens",
+			WebsiteDomain: "supertokens.io",
+		},
+		RecipeList: []supertokens.Recipe{
+			session.Init(&sessmodels.TypeInput{
+				CookieSameSite: &cookieSameSite0,
+			}),
+		},
+	}
+	BeforeEach()
+	StartUpST("localhost", "8080")
+	err := supertokens.Init(configValue)
+	if err != nil {
+		t.Errorf(err.Error())
+	}
+
+	defer AfterEach()
+}
+
+func TestSuperTokensInitWithConfigForSessionModulesWithSameSiteValueAsStrict(t *testing.T) {
+	cookieSameSite0 := " STRICT "
+	configValue := supertokens.TypeInput{
+		Supertokens: &supertokens.ConnectionInfo{
+			ConnectionURI: "http://localhost:8080",
+		},
+		AppInfo: supertokens.AppInfo{
+			APIDomain:     "api.supertokens.io",
+			AppName:       "SuperTokens",
+			WebsiteDomain: "supertokens.io",
+		},
+		RecipeList: []supertokens.Recipe{
+			session.Init(&sessmodels.TypeInput{
+				CookieSameSite: &cookieSameSite0,
+			}),
+		},
+	}
+	BeforeEach()
+	StartUpST("localhost", "8080")
+	err := supertokens.Init(configValue)
+	if err != nil {
+		t.Errorf(err.Error())
+	}
+
+	defer AfterEach()
+}
+
+func TestSuperTokensInitWithConfigForSessionModulesWithSameSiteValueAsRandom(t *testing.T) {
+	cookieSameSite0 := " random "
+	configValue := supertokens.TypeInput{
+		Supertokens: &supertokens.ConnectionInfo{
+			ConnectionURI: "http://localhost:8080",
+		},
+		AppInfo: supertokens.AppInfo{
+			APIDomain:     "api.supertokens.io",
+			AppName:       "SuperTokens",
+			WebsiteDomain: "supertokens.io",
+		},
+		RecipeList: []supertokens.Recipe{
+			session.Init(&sessmodels.TypeInput{
+				CookieSameSite: &cookieSameSite0,
+			}),
+		},
+	}
+	BeforeEach()
+	StartUpST("localhost", "8080")
+	err := supertokens.Init(configValue)
+	if err != nil {
+		if err != nil {
+			errorMessage := err.Error()
+			if errorMessage != `cookie same site must be one of "strict", "lax", or "none"` {
+				t.Errorf(err.Error())
+			} else {
+				assert.Equal(t, errorMessage, `cookie same site must be one of "strict", "lax", or "none"`)
+			}
+		}
+	}
```

Comment:
This style of testing is very strange. Separate them into different tests, or then do init -> check -> init -> check